### PR TITLE
attempt to stabilize the CompressionIntegrationSpec

### LINKED ISSUE #22702: failed: CompressionIntegrationSpec
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/8857/

Failed test: akka.remote.artery.compress.CompressionIntegrationSpec.wrap around

### PR DESCRIPTION
**Pull request by ktoso**
Attempt to make the spec actually test what it's supposed to and make it more stable as well 

Maybe will resolve https://github.com/akka/akka/issues/22702

Will give it another look after food

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala [hunk 1]
```diff
@@ -293,7 +293,7 @@ private[remote] abstract class InboundCompression[T >: Null](
 
   private[this] val cms = new CountMinSketch(16, 1024, System.currentTimeMillis().toInt)
 
-  log.debug("Initializing {} for originUid [{}]", Logging.simpleName(getClass), originUid)
+  log.debug("[inbound-compression] Initializing {} for originUid [{}]", Logging.simpleName(getClass), originUid)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: better to create the logger correctly, it's currently wrong class
```
val _inboundActorRefsLog = Logging(system, classOf[InboundManifestCompression])
```

**Review by patriknw — COMMENTED** [review]
ok, let me know when it's ready for final review

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala (hunk 1 above)
Comment (ktoso) [reply akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: Good point :)

Comment (ktoso) [reply akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: it's fine actually, no need to change; was just more helpful for me to look for those markers when debugging

**Comment by ktoso** [comment]
This can be reviewed

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala [hunk 2]
```diff
@@ -166,7 +166,7 @@ private[remote] final class InboundActorRefCompression(
   originUid:      Long,
   inboundContext: InboundContext,
   heavyHitters:   TopHeavyHitters[ActorRef])
-  extends InboundCompression[ActorRef](log, settings, originUid, inboundContext, heavyHitters) {
+  extends InboundCompression[ActorRef](log, settings, originUid, inboundContext, heavyHitters) { // FIXME fix the logger
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: was it fixed?

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala [hunk 3]
```diff
@@ -404,7 +405,7 @@ private[remote] abstract class InboundCompression[T >: Null](
               resendCount = 0
               advertiseCompressionTable(association, table)
             } else
-              log.debug("Inbound compression table for originUid [{}] not changed, no need to advertise same.", originUid)
+              log.debug("{} for originUid [{}] not changed, no need to advertise same.", Logging.simpleName(tables.activeTable), originUid)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: is this fixed by the logger instead?

File: akka-remote/src/test/scala/akka/remote/artery/compress/CompressionIntegrationSpec.scala [hunk 4]
```diff
@@ -280,35 +284,57 @@ class CompressionIntegrationSpec extends ArteryMultiNodeSpec(CompressionIntegrat
...
+        iteration += 1
+        val echoWrap = createAndIdentify(iteration) // create a different actor for every iteration
+        allRefs ::= echoWrap
+
+        // cause echo to become a heavy hitter
+        (1 to messagesToExchange).foreach { i ⇒ echoWrap ! TestMessage("hello") }
+        receiveN(messagesToExchange) // the replies
+
+        var currentTable: CompressionTable[ActorRef] = null
+        receivedActorRefCompressionTableProbe.awaitAssert {
```
Comment (patriknw) [inline akka-remote/src/test/scala/akka/remote/artery/compress/CompressionIntegrationSpec.scala]: what is the remaining timeout for this probe? how long should it retry the awaitAssert?

**Review by patriknw — COMMENTED** [review]
I'll add this commit to my repeat job

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala (hunk 2 above)
Comment (ktoso) [reply akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: Yeah, the logger is fine actually; was just blind when debugging when I added those prefixes

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala (hunk 3 above)
Comment (ktoso) [reply akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: Yes the logger was fine actually, my bad

Comment (patriknw) [reply akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: but the logger is still wrong, ActorRef vs Manifest:
```
val _inboundActorRefsLog = Logging(system, classOf[InboundManifestCompression])
```

Comment (ktoso) [reply akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: Thanks, fixed

File: akka-remote/src/test/scala/akka/remote/artery/compress/CompressionIntegrationSpec.scala (hunk 4 above)
Comment (ktoso) [reply akka-remote/src/test/scala/akka/remote/artery/compress/CompressionIntegrationSpec.scala]: Good catch, it seems it reaches into the 3 minutes timeout from the outer within. I'll put a specific max here

Comment (ktoso) [reply akka-remote/src/test/scala/akka/remote/artery/compress/CompressionIntegrationSpec.scala]: Ah it'd be bounded by the expectMsgType being 10 seconds internally anyway, but I'll make it explcitic

**Comment by ktoso** [comment]
Addressed review and @patriknw had it running in a repeat job.
Good to merge?

@akka/akka-team

**Comment by ktoso** [comment]
Failure was https://github.com/akka/akka/issues/23449

**Comment by ktoso** [comment]
PLS BUILD

**Pull request merged**